lib/scenic/statements.rb

```diff
@@ -141,7 +159,9 @@ def replace_view(name, version: nil, revert_to_version: nil, materialized: false
         raise ArgumentError, "Cannot replace materialized views"
       end
 
-      sql_definition = definition(name, version)
+      defined_as ||= name
```

Comment:
here we don't raise the same error as above: if that is on purpose, perhaps two methods named like `set_definition` and `set_definition!` (is that the Rubyism? I feel like I'm trying to use "extra" correctly at the moment...)